src/plugins/hyperbee/hyperbee.cpp

```diff
@@ -0,0 +1,1247 @@
+        g_free(write_trap);
+        return nullptr;
+    }
+
+    fault_data_new->plugin = fault_data_old->plugin;
+    fault_data_new->page_va = fault_data_old->page_va;
+
+    //Set the type of the trap.
+    write_trap->type = MEMACCESS;
+    //Guest page-frame number to set event (as defined in events.h) (This is the level 1 translation physical address of the guest. The windows used physical address).
+    write_trap->memaccess.gfn = info->trap->memaccess.gfn;
+    write_trap->memaccess.type = POST; //Do something after sth was written
+    write_trap->memaccess.access = VMI_MEMACCESS_W; //When memory shall be written
+    write_trap->data =
+        fault_data_new; //Use new allocated fault_data_struct to prevent memory leaks and unexpected behaviour in case of memory corruptions
+
+    //Cb is the asynchronous call back https://github.com/tklengyel/drakvuf/issues/1056#issuecomment-713867399
+    write_trap->cb = write_faulted_cb;
+    write_trap->name = "write_faulted_cb";
+    return write_trap;
+}
+
+/**
+ * Creates an execute trap, whose pointer is returned.
+ *
+ * @param info
+ * @param fault_data_old
+ */
+drakvuf_trap_t* create_execute_trap(addr_t gfn, fault_data_struct* fault_data_old)
+{
+
+    auto* exec_trap = (drakvuf_trap_t*) g_try_malloc0(sizeof(drakvuf_trap_t));
+    if (!exec_trap)
+    {
+        return nullptr;
+    }
+
+    //Create a new struct for exec_fault_data and reserve memory.
+    auto* fault_data_new = (struct fault_data_struct*) g_try_malloc0(sizeof(struct fault_data_struct));
+    if (!fault_data_new)
+    {
+        //If there is a problem with allocation of fault_data_new but exec_trap was allocated successfully, free memory of write_trap.
+        g_free(exec_trap);
+        return nullptr;
+    }
+
+    fault_data_new->plugin = fault_data_old->plugin;
+    fault_data_new->page_va = fault_data_old->page_va;
+
+    //Set the type of the trap.
+    exec_trap->type = MEMACCESS;
+    //Guest page-frame number to set event (as defined in events.h) (This is the level 1 translation physical address of the guest. The windows used physical address).
+    exec_trap->memaccess.gfn = gfn;
+    exec_trap->memaccess.type = PRE; //Do something before the access
+    exec_trap->memaccess.access = VMI_MEMACCESS_X; //When memory shall be executed
+    exec_trap->data =
+        fault_data_new; //Use new allocated fault_data_struct to prevent memory leaks and unexpected behaviour in case of memory corruptions
+
+    //Cb is the asynchronous call back https://github.com/tklengyel/drakvuf/issues/1056#issuecomment-713867399
+    exec_trap->cb = execute_faulted_cb;
+    exec_trap->name = "execute_faulted_cb";
+    return exec_trap;
+}
+
+/**
+ * If an instruction fetch got detected on a monitored page, this callback is executed
+ *
+ */
+static event_response_t execute_faulted_cb(drakvuf_t drakvuf, drakvuf_trap_info_t* info)
```

Comment:
this is a 500LoC function - please chop it into smaller functions.

Reply:
Working on that.

Reply:
Done. Was quite hard to do that large refactor and I've to test if all frees and stuff work but the main functionality is still included